File: python/5_kyu/weight_for_weight.py
# Weight for Weight
# https://www.codewars.com/kata/55c6126177c9441a570000cc/train/python

def order_weight(list_of_weights):
  # split the list into a list of integers
  list_of_weights = [i for i in list_of_weights.split(" ")]
  summed_weights = []

  for i in list_of_weights:
    digits = [int(j) for j in list(i)]
    sum_digits = sum(digits)

    summed_weights.append(sum_digits)
  
  return summed_weights

print(order_weight("103 123 4444 99 2000"))
print( order_weight("2000 10003 1234000 44444444 9999 11 11 22 123"))

File: javascript/5_kyu/weight_for_weight.js
// Weight for Weight
// https://www.codewars.com/kata/55c6126177c9441a570000cc/train/javascript

function sumDigits(s) {
    return s.split("").reduce(function(a, b) { return a + parseInt(b) }, 0);
}

function orderWeight(string_of_weights) {
    return string_of_weights.split(" ").sort(function(a, b) {
        return sumDigits(a) - sumDigits(b) || a.localeCompare(b);
    }).join(" ");
}

console.log(orderWeight("103 123 4444 99 2000"));
console.log(orderWeight("2000 10003 1234000 44444444 9999 11 11 22 123"));

// Note: Harder than usual challenge. Study thoroughly